### PR TITLE
Add filter ionmode

### PR DESCRIPTION
**Pull request by niekdejonge**
Added a filter function to remove the wrong ionmode

### CODE REVIEW

File: matchms/filtering/require_correct_ionmode.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+def require_correct_ionmode(spectrum_in: Spectrum,
+                            ion_mode_to_keep):
+    """Removes spectra that are not in the correct ionmode"""
+    if spectrum_in is None:
+        return None
+    spectrum = spectrum_in.clone()
+
+    assert ion_mode_to_keep in {"positive", "negative", "both"}, \
+        "ion_mode should be set to 'positive', 'negative' or 'both'"
```
Comment (hechth) [inline matchms/filtering/require_correct_ionmode.py]: assert is generally not advised to use in normal code I think - the recommended thing is to throw an exception. (https://stackoverflow.com/questions/944592/best-practice-for-using-assert) asserts might also get removed for optimization, so they are not a very stable way of handling errors

Comment (hechth) [reply matchms/filtering/require_correct_ionmode.py]: We could also start having a collection of constants as enums or so for these categorical variables.

File: tests/filtering/test_require_corect_ionmode.py
Comment (hechth) [inline tests/filtering/test_require_corect_ionmode.py:1]: There is a typo in the filename!

File: tests/filtering/test_require_corect_ionmode.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ("positive", "positive", False),
+    ("negative", "negative", False),
+    ("positive", "both", False),
+    ("negative", "both", False),
+    ("positive", "negative", True),
+    ("negative", "positive", True),
+    ("n/a", "both", True),
+])
+def test_clean_compound_name(ionmode, ionmode_to_keep, spectrum_removed):
+    builder = SpectrumBuilder()
```
Comment (hechth) [inline tests/filtering/test_require_corect_ionmode.py]: I think this test should be renamed as it doesn't test cleaning the compound name?

**Review by hechth — CHANGES_REQUESTED** [review]
Please see my comments about the test names and filenames, otherwise seems good.

File: tests/filtering/test_require_corect_ionmode.py
Comment (niekdejonge) [reply tests/filtering/test_require_corect_ionmode.py:1]: Well spotted :)

File: tests/filtering/test_require_corect_ionmode.py (hunk 2 above)
Comment (niekdejonge) [reply tests/filtering/test_require_corect_ionmode.py]: Yes! Copied an old test as scaffold...

File: matchms/filtering/require_correct_ionmode.py (hunk 1 above)
Comment (niekdejonge) [reply matchms/filtering/require_correct_ionmode.py]: The collection of constants might be good to add at some point indeed. I will leave it like this for now.

**Pull request merged**